From ac927cb96d21d1d4ee96be835c60b92cea97fb00 Mon Sep 17 00:00:00 2001
From: Justin Maggard <jmaggard10@gmail.com>
Date: Fri, 20 Feb 2015 00:17:42 -0800
Subject: [PATCH 2/3] rtnl: Ignore temporary cloned routes

Whenever the kernel does a IPv6 route lookup, we get a netlink
"new route" notification for the destination with the RTM_F_CLONED
flag set to note that this is a temporary route.

Let's ignore these messages, or else our log may get filled with
tons of add/del route noise.
---
 src/rtnl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/rtnl.c b/src/rtnl.c
index e03a390..634b523 100644
--- a/src/rtnl.c
+++ b/src/rtnl.c
@@ -1083,6 +1083,8 @@ static void rtnl_route(struct nlmsghdr *hdr)
 
 static bool is_route_rtmsg(struct rtmsg *msg)
 {
+	if (msg->rtm_flags & RTM_F_CLONED)
+		return false;
 
 	if (msg->rtm_table != RT_TABLE_MAIN)
 		return false;
-- 
1.9.1

